### ISSUE TITLE
[Task]: Improve Evaluation + Linting performance

### ISSUE THREAD

**Issue by Rishabh-Rathod** [appsmithorg/appsmith#16259]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Why work on this?
- Recently, we have got a lot of user complaints about the linting feedback cycle being slow on JSEditor
- For 1st evaluation, `lintTree` method takes ~34% time ( depending on the application's JS code this could vary ) once we separate the linting process from the evaluation core flow we could show data on the page faster. ( 34% )
( numbers were derived from profile shared [here](https://theappsmith.slack.com/archives/C02K0SZQ7V3/p1662536902927249?thread_ts=1662451021.891779&cid=C02K0SZQ7V3) )

![image (3)](https://user-images.githubusercontent.com/23132741/189113917-c4ee60ad-f042-4980-ab0f-bc3669ee6129.png)

### Tasks
- [ ] Separate linting worker thread from a current evaluation worker thread to get a response faster and run evaluation and lining in parallel
- [ ] move the errors stored from the data tree evaluation path to a separate structure.

**Comment by ohansFavour** [comment]
Relevant [doc](https://www.notion.so/appsmith/Splitting-Evaluation-and-Linting-5c5d2d9bab164f9eaa9bd4d8933a01dc)

**Comment by billyrudi** [comment]
im interesting in this. i have the same slows pro
blems!

**Comment by samg666** [comment]
I have the same issue. The page loading takes about 8 seconds even without any queries run on pageLoad. 

![iShot_2022-09-26_15 37 53](https://user-images.githubusercontent.com/29649651/192460549-69c90fc3-8d17-4429-b0e1-0eb87640afa9.gif)

**Comment by billyrudi** [comment]
Hi, do you have news about this task? the speed i one of the most important target in every software today!

**Comment by billyrudi** [comment]
please make this with higth priority!

**Comment by Rishabh-Rathod** [comment]
> _do you have news about this task? the speed i one of the most important target in every software today!_

@billyrudi Performance is indeed our priority too. This is a high-priority issue and the team has already picked it up and started initial discussions on this. We will share more updates on this soon. :) 

cc: @satbir121 @bharath31

**Comment by billyrudi** [comment]
Hi, do you have news?

**Comment by satbir121** [comment]
@billyrudi We are actively working on this. It should be resolved before mid Oct. We are expecting around 40% increase in evaluations performance and around 30% increase in linting performance.

**Comment by billyrudi** [comment]
this solves also this issue? ->  https://github.com/appsmithorg/appsmith/issues/16259#issuecomment-1259092110

**Comment by satbir121** [comment]
@billyrudi This issue will help by improving the Page Load performance by around 10-20% depending upon the number of widgets being used.

However, table performance is a separate issue. We are actively working on it. We will share the timelines soon.

**Comment by satbir121** [comment]
@billyrudi Issues that will have major impact on Table performance:
#17179 #16567

**Comment by samg666** [comment]
Hello, does this issue has been improved? I have upgraded to v1.8.8, but seem still no improvements.

![iShot_2022-11-21_11 35 58](https://user-images.githubusercontent.com/29649651/202960592-66f665cc-fc23-46d4-919d-eb8807004473.png)
